Assets/Scripts/Audio/AudioManager.cs

```diff
@@ -0,0 +1,120 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+public class AudioManager : MonoBehaviour
+{
+    // Singleton pattern
+    public static AudioManager Instance { get; private set; }
+    private void Awake()
+    {
+        if (Instance != null && Instance != this)
+        {
+            Destroy(this);
+        }
+        else
+        {
+            Instance = this;
+            DontDestroyOnLoad(this);
+        }
+    }
+
+    // Consts
+    const string resourcesPath = "Assets/Resources/";
+    const string backgroundMusicPath = "BackgroundMusics/";
+    const string soundFXPath = "SoundFX/";
+
+    // Volume settings
+    private float _backgroundMusicVolume = 1f;
+    private float _soundFXVolume = 1f;
+    private float _masterVolume = 1f;
```

Comment:
Can we move all of these consts into a ScriptableObject, and then just pass the ScriptableObject instance to this MonoBehaviour? That way we can configure these values independently of any AudioManager instance, as well as have the changes propagate across all instances immediately